docs/docs/proposals/003-grafanaserviceaccount-crd.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "GrafanaServiceAccount CRD"
+linkTitle: "GrafanaServiceAccount CRD"
+---
+
+## Summary
+
+Add a GrafanaServiceAccount CRD so the operator can create Grafana Service Accounts using CRs to automate the setup of Grafana Service Accounts in newly setup grafana instances. 
+
+Today its required to manually set them up in a running grafana instance using the Grafana GUI or the HTTP-API. This document introduces the suggestion of having them as separate objects that can be setup by the operator on deploy.
+
+The suggested new features are:
+* Let the operator create a grafana service account using a GrafanaServiceAccount CRD.
+* Let the operator store the token as a k8s-secret
+
+## Info
+
+Status: Suggested
+
+## Motivation
+
+Today Grafana Service Accounts has to be created after deploy when the grafana is running using the HTTP-API or the GUI. I instead suggest to have a Grafana Service Account CRD so that the Service Accounts could be predefined and created by the operator at deploy and the tokens will be created as k8s-secrets that then can be read by applications when needed.
+
+## Verification
+
+- Create integration tests for the operator creating grafana service accounts from a bare minimum yaml
+- Create integration tests for the operator creating grafana service accounts from a fully specified yaml
+- Create integration tests to check that it can rotate/invalidate tokens with TTL set (and passed).
+
+## Current solution
+
+Currently you are only able to create these grafana service accounts using the grafana GUI or by using the HTTP-API after the grafana has already been deployed and is running. And its removed when the grafana pod is restarted/redeployed without persistent storage. Meaning a new service account has to be manually created and its new token has to be updated where its being used.
+
+## Proposal
+
+My proposal is to handle grafana service account as a separate CRD that can be specified by the user even before setup and that can be included in a CICD pipeline. It will enable so that the operator can create predefined service accounts on deploy and store the token in a k8s-secret readable by other applications without any manual steps.
+
+### Defining what Grafana Service Account belongs to what grafana.
+I suggest that its done in the same way as dashboards and datasources are using instanceSelector to find the proper grafana, the same thing could be done with the grafana service accounts using the same labels etc.
+
+### Defining Grafana Service Account to Grafana operator.
+
+Today the Grafana Service Account is only held in memory if not using persistent, so when a pod is restarted or redeployed the grafana service account is removed. The grafana service accounts are also only possible to create using the Grafana GUI or with the HTTP-API and cannot be pre-defined before deploy or kept as IAC.
+
```

Comment:
Bear in mind that the service account token will change once the grafana instance is redeployed (when the pod terminates), so the token for a specific instance might change (not often, but it's something to keep in mind)

Reply:
I dont see an issue with this as long as the k8s-secret is updated with the new token, then other applications that needs the token could read the updated secret to fetch the updated token.